peas/sensors.py

```diff
@@ -167,3 +193,115 @@ def capture(self, use_mongo=True, send_message=True):
             self.logger.debug('No sensor data received')
 
         return sensor_data
+
+
+def auto_detect_arduino_devices(comports=None, logger=None):
+    if comports is None:
+        comports = find_arduino_devices()
+    if not logger:
+        logger = get_root_logger()
+    result = []
+    for port in comports:
+        v = auto_detect_port(port, logger)
+        if v:
+            result.append(v)
+    return result
+
+
+def find_arduino_devices():
+    """Find devices (paths or URLs) that appear to be Arduinos.
+
+    Returns:
+        a list of strings; device paths (e.g. /dev/ttyACM1) or URLs (e.g. rfc2217://host:port
+        arduino_simulator://?board=camera).
+    """
+    comports = list_comports()
+    return [p.device for p in comports if 'Arduino' in p.description]
+
+
+def auto_detect_port(port, logger):
+    """Open a port and determine which type of board its producing output.
+
+    Returns: (name, serial_reader) if a recognizable device is connected, else None.
+    """
+    logger.debug('Attempting to connect to serial port: {}'.format(port))
+    try:
+        serial_reader = SerialData(port=port)
```

Comment:
If I add a `baudrate=9600` here then your script works. I think. I just have an Uno but nothing hooked up to it, so I get the json response but have `nan` for all values so it doesn't parse. But that is different than with no baudrate in which case I see the null arrays.